src/main/java/org/spongepowered/common/data/processor/data/entity/SizeDataProcessor.java

```diff
@@ -0,0 +1,89 @@
+ *
+ * Copyright (c) SpongePowered <https://www.spongepowered.org>
+ * Copyright (c) contributors
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package org.spongepowered.common.data.processor.data.entity;
+
+import static org.spongepowered.common.data.util.DataUtil.getData;
+
+import com.google.common.collect.ImmutableMap;
+import net.minecraft.entity.Entity;
+import org.spongepowered.api.data.DataContainer;
+import org.spongepowered.api.data.DataHolder;
+import org.spongepowered.api.data.DataTransactionBuilder;
+import org.spongepowered.api.data.DataTransactionResult;
+import org.spongepowered.api.data.key.Key;
+import org.spongepowered.api.data.key.Keys;
+import org.spongepowered.api.data.manipulator.immutable.entity.ImmutableSizeData;
+import org.spongepowered.api.data.manipulator.mutable.entity.SizeData;
+import org.spongepowered.common.data.manipulator.mutable.entity.SpongeSizeData;
+import org.spongepowered.common.data.processor.common.AbstractEntityDataProcessor;
+
+import java.util.Map;
+import java.util.Optional;
+
+public class SizeDataProcessor extends AbstractEntityDataProcessor<Entity, SizeData, ImmutableSizeData> {
+
+    public SizeDataProcessor() {
+        super(Entity.class);
+    }
+
+    @Override
+    public Optional<SizeData> fill(DataContainer container, SizeData sizeData) {
+        sizeData.set(Keys.HEIGHT, getData(container, Keys.HEIGHT));
+        sizeData.set(Keys.BASE_SIZE, getData(container, Keys.BASE_SIZE));
+        sizeData.set(Keys.SCALE, getData(container, Keys.SCALE));
+        return Optional.of(sizeData);
+    }
+
+    @Override
+    public DataTransactionResult remove(DataHolder dataHolder) {
+        return DataTransactionBuilder.failNoData();
+    }
+
+    @Override
+    protected SizeData createManipulator() {
+        return new SpongeSizeData();
+    }
+
+    @Override
+    protected boolean doesDataExist(Entity entity) {
+        return !entity.isDead;
```

Comment:
I'm also not sure that this should be supported for all entities. It only makes sense on certain entites, and in those cases, it's usually only the scale which can be modified client-side.